src/tests/ftest/util/dmg_utils.py

```diff
@@ -1,52 +1,90 @@
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from __future__ import print_function
+
+from command_utils import CommandWithParameters
+from command_utils import BasicParameter, FormattedParameter
+from avocado.utils import process
+
+
+class DmgCommand(CommandWithParameters):
+    """Defines a object representing a dmg (or daos_shell) command."""
+
+    def __init__(self):
+        """Create a dmg Command object."""
+        super(DmgCommand, self).__init__("daos_shell")
+
+        self.request = BasicParameter("{}")
+        self.action = BasicParameter("{}")
+
+        # daos_shell options
+        self.hostlist = FormattedParameter("-l {}")
+        self.hostfile = FormattedParameter("-f {}")
+        self.configpath = FormattedParameter("-o {}")
+
+        # dmg options
+        self.gid = FormattedParameter("--gid={}")
+        self.uid = FormattedParameter("--uid={}")
+        self.group = FormattedParameter("--group={}")
+        self.mode = FormattedParameter("--mode={}")
+        self.size = FormattedParameter("--size={}")
+        self.nvme = FormattedParameter("--nvme={}")
+        self.svcn = FormattedParameter("--svcn={}")
+        self.target = FormattedParameter("--target={}")
+        self.force = FormattedParameter("--force", False)
+        self.pool = FormattedParameter("--pool={}")
+        self.svc = FormattedParameter("--svc={}")
+        self.rank = FormattedParameter("--rank={}")
+        self.cont = FormattedParameter("--cont={}")
+        self.oid = FormattedParameter("--oid={}")
+
+    def get_param_names(self):
+        """Get a sorted list of daos_server command parameter names."""
+        names = super(DmgCommand, self).get_attributes(FormattedParameter)
+        names.extend(["request", "action"])
+        return names
+
+    def run(self, timeout=None, verbose=True, env=None, shell=False,
+            sudo=False):
+        """ Run the dmg command.
+
+        Args:
+            timeout (int, optional): timeout in seconds. Defaults to None.
+            verbose (bool, optional): display command output. Defaults to True.
+            env (dict, optional): env for the command. Defaults to None.
+            sudo (bool, optional): sudo will be prepended to the command.
+                Defaults to False.
+            subshell (bool, optional): Whether to run cmd on a subshell.
```

Comment:
This doesn't align w/ the method arguments.

Reply:
Will change.